### ISSUE TITLE
how to remove a nested key in an object?

### ISSUE THREAD

**Issue by cultofmetatron** [rethinkdb/rethinkdb#2608]
TLDR: how do I explicitly delete a nested field in an object?

I have a users table, the row for that has a nested field 'channels' which may or may not have a nested field where each field is a channel name and the value is a permissions object

the idea is that I have a  set of resources which are channels to pubnub streams  for a company that I may assign to users and I'm building a company class with methods to manage the streams, I also need to then remove the channel from the user

to make this more explicit, here's an example

``` javascript
{
  'channels': {
       'uuid-2975696087656': {
         publish: 0,
         subscribe: 1
       }
  }
}
```

now I want to be able to remove a channel from this company, currently I have some code based on has_fields. the idea is that I user hasFields to get all users that have this field in channels.

```
var query = {};
query.channels = {};
query[channelName]: true;

r.table('users').has_fields(query)
.map(function(users) {
  //?? how do I remove that field
})

```

how do I remove that field in the map?

**Comment by mattdodge** [comment]
To make it specific to your example, this is what worked for me:

```
r.table('users').has_fields(query).replace(
  r.row.without({'channels': {'uuid-2975696087656': true}})
)
```

Basically, you can use the [nested fields syntax](https://www.rethinkdb.com/docs/nested-fields/javascript/) but only on the entire row. You can't dive into the row object and then call replace (apparently).
